src/System.Private.CoreLib/shared/System/Guid.cs

```diff
@@ -189,10 +185,8 @@ public Guid(string g)
             }
 
             var result = new GuidResult(GuidParseThrowStyle.All);
-            if (!TryParseGuid(g, ref result))
-            {
-                throw result.CreateGuidParseException();
```

Comment:
`CreateGuidParseException` is used in `GuidResult` only now. It can be changed to private.

Also, the GuidResult fields that track the type of the exception should not be needed anymore.